src/bundles/SellerRegistration/components/ProductsForm/ProductsForm.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {Form, actions} from 'react-redux-form';
+
+import { required } from '../../../../validators';
+
+import Layout        from '../../../../shared/Layout';
+import BaseForm      from '../../../../shared/form/BaseForm';
+import SubmitForm    from '../../../../shared/form/SubmitForm';
+import ErrorBox      from '../../../../shared/form/ErrorBox';
+import Textfield     from '../../../../shared/form/Textfield';
+import Textarea     from '../../../../shared/form/Textarea';
+import formProps     from '../../../../shared/reduxModules/formPropsSelector';
+
+class ProductsForm extends BaseForm {
+
+  static propTypes = {
+    action: React.PropTypes.string,
+    csrf_token: React.PropTypes.string,
+    form: React.PropTypes.object.isRequired
+  }
+
+  onAdd(e) {
+      e.preventDefault();
+      const {model, createProduct, productsForm} = this.props;
+      createProduct(model, productsForm);
+  }
+
+  onRemove(id, e) {
+      e.preventDefault();
+      const {model, removeProduct} = this.props;
+      removeProduct(model, id);
+  }
+
+  render() {
+    const { action, csrf_token, model, form, buttonText, children, onSubmit, productsForm } = this.props;
+
+    return (
+      <Layout>
+        <header>
+          <h1 tabIndex="-1">Products</h1>
+          <p>To add a product to your profile simply share your product name, product summary and add links to extra information for interested buyers. You can add as many products as you like.</p>
+        </header>
+        <article role="main">
+          <ErrorBox focusOnMount={true} model={model}/>
+          <Form model={model}
+            action={action}
+            method="post"
+            id="products"
+            component={SubmitForm}
+            valid={form.valid}
+            onCustomSubmit={onSubmit}
+          >
+            {csrf_token && (
+              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
+            )}
+
+            {productsForm.products && productsForm.products.map((product, key) => {
```

Comment:
`productForm.products` shouldn't ever be falsy - since you've defined it in your reducer. A better defensive check may be to ensure its an array (which would still capture those falsy cases if they were to happen)?